sdk/cosmosdb/cosmos/src/request/request.ts

```diff
@@ -128,8 +128,13 @@ export async function getHeaders({
   }
 
   if (options.maxIntegratedCacheStalenessInMs && resourceType === ResourceType.item) {
-    headers[Constants.HttpHeaders.DedicatedGatewayPerRequestCacheStaleness] =
-      options.maxIntegratedCacheStalenessInMs;
+    if (typeof options.maxIntegratedCacheStalenessInMs === "number") {
```

Comment:
If its a HTTP standard requirement then a common place to marshal type changes is recommended.

Reply:
options.maxIntegratedCacheStalenessInMs.toString()